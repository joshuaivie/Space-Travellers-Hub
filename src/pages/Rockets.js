import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchRockets } from '../redux/rockets/rockets';
import RocketList from '../components/RocketList';

function Rockets() {
  const dispatch = useDispatch();
  const rocketsState = useSelector((state) => state.rockets);

  useEffect(() => {
    if (!rocketsState.rockets.length) {
      dispatch(fetchRockets());
    }
  }, [dispatch]);

  const { rockets } = rocketsState;

  /* useEffect(() => {
  }, [rocketsState]); */
  return (
    <>
      <RocketList rockets={rockets} />
    </>
  );
}

export default Rockets;
