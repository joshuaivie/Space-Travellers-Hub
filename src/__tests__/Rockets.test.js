import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import {
  render, fireEvent, screen,
} from '@testing-library/react';
import Rockets from '../pages/Rockets';
import store from '../redux/configureStore';

describe('It renders correctly', () => {
  it('Renders the rockets page', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Provider store={store}>
            <Rockets />
          </Provider>
        </MemoryRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders the page correctly', () => {
    render(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    );

    expect(screen.getByText(/Reserve Rocket/)).toBeInTheDocument();
  });

  // test('User can reserve rocket', () => {
  //   render(<Calculator />);
  //   const display = screen.getByText('00');
  //   const button = screen.getByText('8');
  //   fireEvent.click(button);
  //   expect(display.innerHTML).toBe('8');
  // });
});
