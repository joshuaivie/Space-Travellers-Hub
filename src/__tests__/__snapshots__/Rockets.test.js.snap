// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders correctly Renders 1`] = `null`;

exports[`It renders correctly Renders the rockets page 1`] = `null`;
